pkg/api/postage_test.go

```diff
@@ -208,6 +208,8 @@ func TestPostageGetStamps(t *testing.T) {
 	cs := &postage.ChainState{Block: 10, TotalAmount: big.NewInt(5), CurrentPrice: big.NewInt(2)}
 	bs := mock.New(mock.WithChainState(cs), mock.WithBatch(b))
 	ts, _, _, _ := newTestServer(t, testServerOptions{DebugAPI: true, Post: mp, BatchStore: bs, BlockTime: big.NewInt(2)})
+	bsForNonExistingStamp := mock.New(mock.WithChainState(cs))
+	tsForNonExistingStamp, _, _, _ := newTestServer(t, testServerOptions{DebugAPI: true, Post: mp, BatchStore: bsForNonExistingStamp, BlockTime: big.NewInt(2)})
```

Comment:
This test is still not correct. We dont use multiple servers for the same test. Secondly, your All stamps test is not different from the single stamp test case. I think its important to understand what you actually want to test before writing the tests.

The thing we NEED to test is that the handler returns only valid stamps (i.e. stamps are are not expired) in normal case and all stamps (even the expired ones) using the argument. So you should have 2 batches, one of them should be expired and one of them should be valid. This way you can query the same server with and without the arguments and be sure that your logic is correct. Current test looks like a hack to get it working.

Reply:
Thank you, lemme fix it, got it now.